src/lib/rng/auto_rng/auto_rng.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+* (C) 2016 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/auto_rng.h>
+#include <botan/entropy_src.h>
+#include <botan/loadstor.h>
+#include <botan/internal/os_utils.h>
+
+#if defined(BOTAN_HAS_HMAC_DRBG)
+  #include <botan/hmac_drbg.h>
+#endif
+
+#if defined(BOTAN_HAS_HMAC_RNG)
+  #include <botan/hmac_rng.h>
+#endif
+
+#if defined(BOTAN_HAS_SYSTEM_RNG)
+  #include <botan/system_rng.h>
+#endif
+
+namespace Botan {
+
+AutoSeeded_RNG::AutoSeeded_RNG(size_t max_output_before_reseed)
+   {
+   m_rng.reset(new BOTAN_AUTO_RNG_DRBG(MessageAuthenticationCode::create(BOTAN_AUTO_RNG_HMAC),
+#if defined(BOTAN_HAS_SYSTEM_RNG)
+                                       system_rng(),
+#else
+                                       Entropy_Sources::global_sources(),
+#endif
```

Comment:
In that particular case, couldn't you just use the es_capi entropy source instead of the System_RNG?

But I see your point now, only relying on CryptGenRandom on Windows feel kind of shaky, because it's closed source. However, libressl and boringssl both do it. Hm.